laravel/app/AutoAssignment.php

```diff
@@ -117,18 +116,15 @@ private static function assignResidentsForPref($residents, $prefNum, $ticketsToA
             $notOnRotation = array(); 
             // identify which residents have/don't have same rotation as the schedule rotation 
             foreach($competingOptions as $competingOption){
-                $resName = Resident::where('id', $competingOption['resident'])
-                            ->value('name');
-                $resRotations = Rotations::where('name', $resName)
-                                ->get();
+                $resName = Resident::find($competingOption['resident'])->value('name');
+                $resRotations = Rotations::where('name', $resName)->get();
                 $serviceId ="";
                 foreach($resRotations as $resRotation){
-                    if($date >= $resRotation['Start'] && $date <= $resRotation['End']){
+                    if($resRotation['Start'] <= $date && $date <= $resRotation['End']){
                         $serviceId = $resRotation['Service'];
                     }
                 }
-                $resRotation = EvaluationForms::where('id', $serviceId)
-                                ->value('rotation');
+                $resRotation = EvaluationForms::where('id', $serviceId)->value('rotation');
```

Comment:
can this be a `find`?

Reply:
No, it is done this way to buffer serviceId not being set / a resident not being in the rotations table at all - this has to do with the living defect in prod

Reply:
Since I'm not familiar with the defect, I don't understand what you mean by this.